node/lib/util/write_repo_ast_util.js

```diff
@@ -399,6 +400,24 @@ git checkout-index -a -f
         }
     }
 
+    // set up a rebase if there is one
+
+    if (null !== ast.rebase) {
+        const rebase = ast.rebase;
+        const originalSha = commitMap[rebase.originalHead];
+        const ontoSha = commitMap[rebase.onto];
+        const original = yield NodeGit.AnnotatedCommit.lookup(repo,
+                                                              originalSha);
+        const onto = yield NodeGit.AnnotatedCommit.lookup(repo, ontoSha);
+        yield NodeGit.Rebase.init(repo, original, onto, null, null);
```

Comment:
That method has a weird, overloaded, contract. The documentation for `branch` is: "The terminal commit to rebase, or NULL to rebase the current branch", and for `upstream`: "The commit to begin rebasing from, or NULL to rebase all reachable commits".